torchvision/datasets/voc.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import print_function
+import os
+import os.path
+import sys
+import torch
+import torch.utils.data as data
+from PIL import Image, ImageDraw
+if sys.version_info[0] == 2:
+    import xml.etree.cElementTree as ET
+else:
+    import xml.etree.ElementTree as ET
+
+VOC_CLASSES = ('background',  # always index 0
+               'aeroplane', 'bicycle', 'bird', 'boat',
+               'bottle', 'bus', 'car', 'cat', 'chair',
+               'cow', 'diningtable', 'dog', 'horse',
+               'motorbike', 'person', 'pottedplant',
+               'sheep', 'sofa', 'train', 'tvmonitor')
+
+# for making bounding boxes pretty
+COLORS = ((255, 0, 0, 128), (0, 255, 0, 128), (0, 0, 255, 128),
+          (0, 255, 255, 128), (255, 0, 255, 128), (255, 255, 0, 128))
+
+
+class VOCSegmentation(data.Dataset):
+    """VOC Segmentation Dataset Object
+    input and target are both images
+
+    NOTE: need to address https://github.com/pytorch/vision/issues/9
+
+    Arguments:
+        root (string): filepath to VOCdevkit folder.
+        image_set (string): imageset to use (eg: 'train', 'val', 'test').
+        transform (callable, optional): transformation to perform on the
+            input image
+        target_transform (callable, optional): transformation to perform on the
+            target image
+        dataset_name (string, optional): which dataset to load
+            (default: 'VOC2007')
+    """
+
+    def __init__(self, root, image_set, transform=None, target_transform=None,
+                 dataset_name='VOC2007'):
+        self.root = root
+        self.image_set = image_set
+        self.transform = transform
+        self.target_transform = target_transform
+
+        self._annopath = os.path.join(
+            self.root, dataset_name, 'SegmentationClass', '%s.png')
+        self._imgpath = os.path.join(
+            self.root, dataset_name, 'JPEGImages', '%s.jpg')
+        self._imgsetpath = os.path.join(
+            self.root, dataset_name, 'ImageSets', 'Segmentation', '%s.txt')
+
+        with open(self._imgsetpath % self.image_set) as f:
+            self.ids = f.readlines()
+        self.ids = [x.strip('\n') for x in self.ids]
+
+    def __getitem__(self, index):
+        img_id = self.ids[index]
+
+        target = Image.open(self._annopath % img_id).convert('RGB')
```

Comment:
I don't think we want to load the `target` as `RGB` images, as we will lose the information from the class indices (and will have instead RGB triplets).